pkg/tidbver/tidbver.go

```diff
@@ -28,7 +28,9 @@ func TiDBSupportSecureBoot(version string) bool {
 
 // TiDBSupportUpgradeAPI return if given version of TiDB support upgrade API
 func TiDBSupportUpgradeAPI(version string) bool {
-	return semver.Compare(version, "v7.4.0") >= 0 || strings.Contains(version, "nightly")
+	return semver.Compare(version, "v7.4.0") >= 0 ||
+		(semver.MajorMinor(version) == "v7.1" && semver.Compare(version, "v7.1.2") >= 0) ||
```

Comment:
Oh, probably not.
For example, upgrade from v7.1.2 to v7.4. However v7.4 does not support the upgrade function.